test/integ/quote-classic.test.ts

```diff
@@ -101,7 +107,32 @@ const checkQuoteToken = (
     ? tokensQuoted.subtract(tokensSwapped)
     : tokensSwapped.subtract(tokensQuoted);
   const percentDiff = tokensDiff.asFraction.divide(tokensQuoted.asFraction);
-  expect(percentDiff.lessThan(new Fraction(parseInt(SLIPPAGE), 100))).to.be.true;
+  expect(
+    percentDiff.lessThan(new Fraction(parseInt(SLIPPAGE), 100)),
+    `expected tokensQuoted ${tokensQuoted.toExact()} actual tokens swapped ${tokensSwapped.toExact()}`
+  ).to.be.true;
+};
+
+const checkPortionRecipientToken = (
+  before: CurrencyAmount<Currency>,
+  after: CurrencyAmount<Currency>,
+  expectedPortionAmountReceived: CurrencyAmount<Currency>
+) => {
+  const actualPortionAmountReceived = after.subtract(before);
+
+  const tokensDiff = expectedPortionAmountReceived.greaterThan(actualPortionAmountReceived)
+    ? expectedPortionAmountReceived.subtract(actualPortionAmountReceived)
+    : actualPortionAmountReceived.subtract(expectedPortionAmountReceived);
+  // There will be a slight difference between expected and actual due to slippage during the hardhat fork swap.
+  const percentDiff = tokensDiff.asFraction.divide(expectedPortionAmountReceived.asFraction);
+  // We can be very very strict here. Slippage tolerance is 5%, but we can make the difference between expected and actual to be within 0.05%,
+  // due to the fact the test setups are swapping the very high liquidity pairs with deep market depth, with fairly small input token amount.
+  expect(percentDiff.lessThan(new Fraction(parseInt(SLIPPAGE), 10000))).to.be.true;
```

Comment:
I think 0.05% is too low here, since these are integ tests it's fine to just do 5% so the tests aren't as flaky and the condition is still being tested.

Reply:
totally agree here btw, i was gonna comment on it too, but i am not an expert so glad Eric brought it up